### PR TITLE
Provide a new "Delete Account" Menu Item

### PR DESCRIPTION
**Pull request by jisqyv**
This action will remove an account provided that there are no
non-trashed projects in the account. If there are, a dialog is
produced explaining that all projects must be delete first.

If there are not projects, then the remaining vestiges of the account
are removed, along with any projects that are marked as trash.

Change-Id: I9c43a4d59a2b49adcafbbe000f1b4a410e727d38

### CODE REVIEW

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 1]
```diff
@@ -765,8 +765,20 @@
   @Description("Label of the link for signing out")
   String signOutLink();
 
+  @DefaultMessage("<font color=red>Delete Account</font>")
+  @Description("Label of the link to delete your account")
+  String deleteAccountLink();
```
[the remark this answers is not in the document]
Comment (jisqyv) [reply appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java:770]: Good Idea

**Comment by ewpatton** [comment]
I believe we have a CSS class for destructive operations. We may be able to leverage that.

**Comment by jisqyv** [comment]
@ewpatton Can we add a style on a menu item?

**Comment by jisqyv** [comment]
@ewpatton So to answer my own question... It can be done, GWT supports it, but we would have to update several a bit of our code in order to have access to the calls we need.

**Comment by jisqyv** [comment]
@ewpatton Turns out it was easier than I thought. Latest commit uses CSS to create Red "Delete Account" menu item

**Comment by SRIKAR-B-S-S** [comment]
I like this Idea @jisqyv !
but I've a doubt that is "_Should we delete all our projects before we delete an account?"_

**Comment by jisqyv** [comment]
@SRIKAR-B-S-S Deleting an account cannot be undone. We require people to delete all of their projects first (they just need to move them to the trash, they don't have to then permanently delete them from the trash) so that people do not delete their account unless they really mean it.

Otherwise, we would have people deleting their account and then changing their mind and asking us to get back their projects, which we won't be able to do...

**Comment by SRIKAR-B-S-S** [comment]
Yeah, Thanks for the clarification 👍

**Comment by jisqyv** [comment]
Note: The support for removing the account from the login server has not been deployed on the login server. If you test this code, you should see an error reporting that the login server returned a 404 (page not found).

**Comment by jisqyv** [comment]
I just removed the top commit (and signed the one below it). I'm going to take a different approach to removing accounts from the login server...

**Comment by jisqyv** [comment]
This change is almost ready. I will squash it down to one commit and un-draft it after I test the login server side to ensure all works as expected.

**Review by SusanRatiLane — APPROVED** [review]

**Pull request merged**